GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -111,11 +81,81 @@ public ToolStripGitStatus()
             _globalIgnoreWatcher.Renamed += GlobalIgnoreChanged;
             _globalIgnoreWatcher.Error += WorkTreeWatcherError;
             _globalIgnoreWatcher.IncludeSubdirectories = false;
-            _workTreeWatcher.NotifyFilter = NotifyFilters.LastWrite;
+            _globalIgnoreWatcher.NotifyFilter = NotifyFilters.LastWrite;
+        }
+
+
+        private GitStatusMonitorState CurrentStatus
+        {
+            get { return _currentStatus; }
+            set
+            {
+                _currentStatus = value;
```

Comment:
I've been thinking whether we should tweak to 

```
if (_currentStatus == value) 
{
    return;
}
_currentStatus = value;
...
```
so it doesn't fire unnecessary events. What do you think?

Reply:
Same status update should only occur at repeated OnUpdateStatusError(), which should not be so common or frequent.
Should change this in the PR for #4069